### ISSUE TITLE
Add an example set-up module

### ISSUE THREAD

**Issue by jbwheatley** [jbwheatley/pact4s#113]
Sending people to the test classes to look at examples isn't that user-friendly, and lots there is obfuscated for the purpose of brevity. Would be better to create a module with a full example set up for a request-response relationship, and for a messaging relationship.

**Comment by gaeljw** [comment]
Do you have something specific in mind @jbwheatley? I'll be happy to help on creating this example project.

It could be a simple http4s or Akka HTTP producer and a sttp client for instance 🤷

**Comment by jbwheatley** [comment]
yeah, I was thinking basically recreate what we have in the tests, but in a more readable format with comments etc

**Comment by jbwheatley** [comment]
have partially done this

**Comment by danlaunl** [comment]
small newbie question: how do i run the example via sbt ? clean; compile; example/test shows (like via intellij) "no tests found" . Also when I remove the lines `      publish / skip := true,
     Test / skip    := true` from build.sbt
     
 btw, i would like to PR a spray-json module and want to make sure module `example`  would also pass/work (also before trying in an own project)

**Comment by jbwheatley** [comment]
hey @danlaunl. The example directory consists of two separate sbt projects, `consumer` and `producer`, which will be why `example/test` is saying no tests found

**Comment by danlaunl** [comment]
so how should i run those tests? i also tried cd 'ing in , and then sbt but that doens't compile. 

or something like this ? https://stackoverflow.com/questions/6362286/how-do-i-get-sbt-0-10-0-to-compile-files-in-a-subdirectory

**Comment by jbwheatley** [comment]
sorry you're completely right, the examples won't run. I've opened an issue #291

**Comment by jbwheatley** [comment]
should be fixed in latest master
